zipline/pipeline/engine.py

```diff
@@ -273,6 +274,25 @@ def _inputs_for_term(self, term, workspace, graph):
             out.append(input_data)
         return out
 
+    def _similar_atomic_terms(self, graph, atomic_term):
+        loader_dispatch = self.loader_dispatch
+        loader = loader_dispatch(atomic_term)
+        extra_rows = graph.extra_rows[atomic_term]
+
+        for term in graph.atomic_terms:
+            if (loader_dispatch(term) == loader
+                    and graph.extra_rows[term] == extra_rows):
+                yield term
+
+    def loader_dispatch(self, term):
+        if term is AssetExists():
+            return None
+
+        loader = self._loader_dispatch(term)
```

Comment:
I feel like `None` should never be a valid return from whatever does the dispatching here.  I'd have the contract be that the dispatcher should raise something like `NoSuchLoader(term)`